Fix build failure with Bazel@HEAD

In https://github.com/bazelbuild/bazel/pull/11665, Bazel upgraded java_tools to v9.0 which has a new error prone version.

The build is failing with:
```
external/io_grpc_grpc_java/core/src/main/java/io/grpc/internal/RetriableStream.java:702: error: [GuardedBy] This access should be guarded by 'this.scheduledHedging.lock'; instead found: 'this.lock'
        futureToBeCancelled = scheduledHedging.markCancelled();
                                              ^
    (see https://errorprone.info/bugpattern/GuardedBy)
```
https://buildkite.com/bazel/bazel-at-head-plus-downstream/builds/1545#9679232e-e81f-4dbd-9b7c-526c9cfd0af2

--- a/WORKSPACE
+++ b/WORKSPACE
@@ -39,14 +39,12 @@ rules_proto_toolchains()
 # BEGIN: gRPC dependencies
 http_archive(
     name = "io_grpc_grpc_java",
-    # TODO: Update reference to a release version (once it exists) that contains
-    # a4299eb6bed3c2f2497f583d0c4620c9f31ec455.
-    # Commit a4299eb6bed3c2f2497f583d0c4620c9f31ec455 fixed
-    # https://github.com/grpc/grpc-java/issues/6536. That bug caused test failures on Bazel CI.
-    # We don't specify sha256, because the sha256 of GitHub-served non-release archives isn't
-    # stable.
-    urls = ["https://github.com/grpc/grpc-java/archive/a4299eb6bed3c2f2497f583d0c4620c9f31ec455.tar.gz"],
-    strip_prefix = "grpc-java-a4299eb6bed3c2f2497f583d0c4620c9f31ec455",
+    urls = [
+        "https://mirror.bazel.build/github.com/grpc/grpc-java/archive/v1.29.0.tar.gz",
+        "https://github.com/grpc/grpc-java/archive/v1.29.0.tar.gz",
+    ],
+    sha256 = "62faefec4c211709416427ce071f66c901b449fbc471ecb03fbf71d0675db4a3",
+    strip_prefix = "grpc-java-1.29.0",
 )
 
 load("@io_grpc_grpc_java//:repositories.bzl", "grpc_java_repositories")

